### PR TITLE
Michaels branch

### PR DESCRIPTION
**Pull request by Michael-Reefe**
Should I keep making new pull requests every time I make new commits, or should we just keep it as one?

### CODE REVIEW

File: main/observing/observation_run.py [hunk 1]
```diff
@@ -15,14 +17,27 @@ def __init__(self, observation_request_list, image_directory):
         '''
         self.image_directory = image_directory
         self.observation_request_list = observation_request_list
-        self.filterwheel_dict = filter_wheel.get_filter().filter_position_dict()
         self.camera = Camera()
-
+        #self.telescope = Telescope()
+        
+        self.filterwheel_dict = {value: int(key.replace('position_', '')) for key,value in
+                                 ObjectReader(Reader(r'c:\users\gmu observtory1\-omegalambda\config\fw_config.json')).ticket.__dict__.items()}
```
Comment (Kakon24) [inline main/observing/observation_run.py]: This path should not be hard coded, it should be passed via the driver.  Eventually we will start using a config file for these arguments.

File: main/observing/observation_run.py [hunk 2]
```diff
@@ -43,17 +57,17 @@ def run_ticket(self, ticket):
...
 
+    def take_images(self, name, num, exp_time, current_filter, end_time, path):
+    
+        image_num = 1
         for i in range(num):
             if end_time <= datetime.datetime.now():
                 print("The observations end time of {} has passed.  "
                       "Stopping observation of {}.".format(end_time, name))
-
-            current_filter = filter[num_filters % i]
```
Comment (Kakon24) [inline main/observing/observation_run.py:57]: So this idea was correct but my implementation is wrong.  It should be filter[i % num_filters]

File: testing_log/02-17-20.txt [hunk 3]
```diff
@@ -0,0 +1,236 @@
+First test run of test_driver.py:
+
```
Comment (Kakon24) [inline testing_log/02-17-20.txt]: Logs should not be merged to dev.

File: main/observing/observation_run.py (hunk 2 above)
Comment (Kakon24) [reply main/observing/observation_run.py:57]: This touches on how we wanted the observation ticket to be read and executed.  The details of which I don't think have been passed on to you.

File: main/common/datatype/object_reader.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from main.common.datatype.observation_ticket import ObservationTicket
+from main.common.datatype.filter_wheel import FilterWheel
+
+Objects = {"observation_ticket": ObservationTicket, "filter_wheel": FilterWheel}
+
+class ObjectReader():
+    
+    def __init__(self, reader_obj=None):
```
Comment (Kakon24) [inline main/common/datatype/object_reader.py]: Get rid of None

**Comment by Kakon24** [comment]
Demoed on 5/26/2020, ready for merge.

**Comment by Michael-Reefe** [comment]
Hi Shawn,

I just noticed that when you merged, the stuff I had in my .gitignore (like
the útil directory for example) was also removed from dev.  Was that
supposed to happen?

Michael Reefe

**Pull request merged**